centreon/src/Core/Dashboard/Application/UseCase/FindPerformanceMetricsData/FindPerformanceMetricsDataRequest.php

```diff
@@ -23,14 +23,60 @@
 
 namespace Core\Dashboard\Application\UseCase\FindPerformanceMetricsData;
 
+use Core\Common\Infrastructure\Validator\DateFormat;
+use Symfony\Component\Validator\Constraints as Assert;
+
 final class FindPerformanceMetricsDataRequest
 {
-    /** @var string[] */
-    public array $metricNames = [];
-
+    /**
+     * @param string $start
+     * @param string $end
+     * @param string[] $metricNames
+     */
     public function __construct(
-        public \DateTimeInterface $startDate,
-        public \DateTimeInterface $endDate,
+        #[Assert\NotBlank]
+        #[Assert\DateTime(
+            format: DateFormat::ISO8601,
+            message: DateFormat::INVALID_DATE_MESSAGE
+        )]
+        public readonly mixed $start,
+        #[Assert\NotBlank]
+        #[Assert\DateTime(
+            format: DateFormat::ISO8601,
+            message: DateFormat::INVALID_DATE_MESSAGE
+        )]
+        public readonly mixed $end,
+        #[Assert\NotNull]
+        #[Assert\Sequentially([
+            new Assert\Type('array'),
+            new Assert\All(
+                [new Assert\Type('string')]
+            ),
+        ])]
+        public readonly mixed $metricNames,
     ) {
     }
+
+    /**
+     * @return FindPerformanceMetricsDataRequestDto
+     */
+    public function toDto(): FindPerformanceMetricsDataRequestDto
+    {
+        return new FindPerformanceMetricsDataRequestDto(
+            startDate: new \DateTimeImmutable($this->start),
+            endDate: new \DateTimeImmutable($this->end),
+            metricNames: $this->sanitizeMetricNames()
+        );
+    }
+
+    /**
+     * @return string[]
+     */
+    private function sanitizeMetricNames(): array
+    {
+        return array_map(
+            static fn (string $metricName): string => \trim($metricName, '"'),
+            $this->metricNames
+        );
+    }
```

Comment:
This is the kind of code that is prompt to duplication as I assume this is not the only place we have to do that.
Therefore, it could be interesting to think to put that at a higher level so it can be shared by other services if needed (statically or through a dedicated service).